Test/Algorithms/DirectSearch/CMA.cpp

```diff
@@ -14,6 +14,51 @@
 
 using namespace shark;
 
+
+struct MultiplicativeNoisySphere : public SingleObjectiveFunction {
+	
+	MultiplicativeNoisySphere(std::size_t numberOfVariables, double sigma):m_numberOfVariables(numberOfVariables), m_sigma(sigma) {
+		m_features |= CAN_PROPOSE_STARTING_POINT;
+		m_features |= IS_NOISY;
+	}
+
+	/// \brief From INameable: return the class name.
+	std::string name() const
+	{ return "MultiplicativeNoisySphere"; }
+
+	std::size_t numberOfVariables()const{
+		return m_numberOfVariables;
+	}
+	
+	bool hasScalableDimensionality()const{
+		return true;
+	}
+
+	void setNumberOfVariables( std::size_t numberOfVariables ){
+		m_numberOfVariables = numberOfVariables;
+	}
+
+	SearchPointType proposeStartingPoint() const {
+		RealVector x(numberOfVariables());
+
+		for (std::size_t i = 0; i < x.size(); i++) {
+			x(i) = Rng::gauss(0,1);
+		}
+		return x;
+	}
+
+	double eval(SearchPointType const& x) const {
+		SIZE_CHECK(x.size() == numberOfVariables());
+		m_evaluationCounter++;
+		double f = norm_sqr(x);
+		double noise = exp(Rng::gauss(0,m_sigma));
```

Comment:
I have changed this and somehow understand why you would like this, as this will make the worst points of the set appear the best when a negative epsilon is drawn. However, i would argue that this test is actually a lot easiuer than the one i came up with: simply because this test is likely to generate more extreme reorderings by single "bad" draws (e.g. worst->best) which the current mechanism is much better in picking up. The result is that the noise estimate is much more stable with your proposed distribution, even though the noise level is much smaller. I have still changed to the first version you proposed.